source/client/options_impl.cc

```diff
@@ -293,6 +303,15 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
     throw MalformedArgvException("Invalid value for --max-requests-per-connection");
   }
 
+  if (!tls_context.getValue().empty()) {
+    ENVOY_LOG(warn, "--tls-context is deprecated. "
+                    "It can be replaced by an equivalent --transport-socket. "
+                    "See --help for an example.");
+  }
+  if (!tls_context.getValue().empty() && !transport_socket.getValue().empty()) {
+    ENVOY_LOG(warn, "Setting both --tls-context and --transport-socket "
+                    "may produce undefined behavior.");
```

Comment:
Drive-by comment: I wonder if it would make sense for us to wire up a way to change the execution status if we observe errors and/or warnings being send to the logs.

Reply:
+1 for a hard error here, sorry about the noise.

Reply:
Done